src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -208,9 +210,15 @@ public class List<T> : IList<T>, System.Collections.IList, IReadOnlyList<T>
         // increased by one. If required, the capacity of the list is doubled
         // before adding the new element.
         //
-        public void Add(T item) {
-            if (_size == _items.Length) EnsureCapacity(_size + 1);
-            _items[_size++] = item;
+        public void Add(T item)
+        {
+            var size = _size;
+            if (size == _items.Length)
+            {
+                EnsureCapacity(size + 1);
```

Comment:
Noinlining should always be honored, it's one of the first things the jit looks for.